compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12845,6 +12845,151 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    }
 
 
+bool canTransformOp(bool canTransformAdd, bool canTransformSub, TR::Node *node, TR::Simplifier *s)
+   {
+   if (!(canTransformAdd || canTransformSub))
+      {
+      if (s->trace())
+         traceMsg(s->comp(),
+                  "\nEliminating add/sub under order compare node n%dn failed due to overflow\n",
+                  node->getGlobalIndex());
+
+      return false;
+      }
+   return true;
+   }
+
+TR::Node *
+arithUnderSignedAndUnsignedCompareHandler(TR::Node *opNode, TR::Node *secondChild, TR::Simplifier *s, TR::Node *node,
+                                          bool isSigned, bool isAddOp, bool isSubOp)
```

Comment:
I think this function deserves a Doxygen comment, especially because it's really not obvious (at least to me) what `opNode`, `secondChild`, and `node` are supposed to be without reading through the code.

Reply:
@Leonardo2718 what do you think? I'm concerned its too similar to the Doxygen comment under removeArithmeticsUnderIntegralCompare